corehq/form_processor/submission_post.py

```diff
@@ -290,21 +308,18 @@ def process_signals(self, instance):
                 ) % (func, instance.form_id, type(resp).__name__, error_message))
                 errors.append(error_message)
         if errors:
-            self.interface.xformerror_from_xform_instance(instance, ", ".join(errors))
+            self.interface.xformerror_from_xform_instance(instance, ", ".join(errors), with_new_id=True)
             self.formdb.update_form_problem_and_state(instance)
+            publish_form_deleted(self.domain, instance.orig_id)
+            publish_form_saved(instance)
```

Comment:
these should be moved to the `FormAccessorSQL` method. We don't need to publish changes for Couch domains since the changes get published from the couch change feed.